### DISCUSSION TITLE
How to use SAM text prompts

### DISCUSSION THREAD

**Discussion by BuKeMod** [opengeos/segment-geospatial#218] [Q&A] [answered]
Hello. Is there a way to use SAM for detecting and segmenting multiple things without detecting the same thing again? I'm using SAM text prompts, but it detects duplicates.

![image](https://github.com/opengeos/segment-geospatial/assets/71030661/c4df7b0d-2dec-4a14-9e01-0d2f26c4c6f1)

![image](https://github.com/opengeos/segment-geospatial/assets/71030661/230dcb9e-62de-421e-a77a-c812d9720fd6)

**Comment by dempsey-ryan** [comment]
Are the two segmentation maps generated from different text prompts?

**Reply by BuKeMod** [reply]
Yes, the maps were generated from different text prompts.

**Reply by dempsey-ryan** [reply]
I am also curious about this. Whether I keep prompts separate or I combine them e.g. "A or B", multiple objects (sometimes both A and B) seem to overlap. I wonder if there is a way to come up with a single map where every pixel in the image has only one label (e.g. A, B, or none)

**Comment by giswqs** [comment, marked as answer]
The text prompt functionality relies on [GroundingDINO](https://github.com/IDEA-Research/GroundingDINO) to detect the objects and generate bounding boxes. The duplicates are likely resulted from GroundingDINO. I don't have a good solution for this issue. YOu probably have to do some post-processing to remove the duplicates. Some simple GIS overlay analysis can identify the duplicates.
